packages/driver/src/cy/multi-domain/index.ts

```diff
@@ -6,6 +6,20 @@ import $errUtils from '../../cypress/error_utils'
 
 export function addCommands (Commands, Cypress: Cypress.Cypress, cy: Cypress.cy, state: Cypress.State, config: Cypress.InternalConfig) {
   let timeoutId
+  const correctStackForCrossDomainError = (error: any, userInvocationStack: string) => {
+    //  Since Errors sent over postMessage need to be serialized to Objects, we need to serialize them back
+    const parsedError = $errUtils.serializeCrossDomainErrorsFromObject(error)
+
+    // With just the error message at hand, attach the switchToDomain user invocation stack to point users
+    // To where the error occurred in their spec file
+    $errUtils.enhanceStack({
+      err: parsedError,
+      userInvocationStack,
+      projectRoot: config('projectRoot'),
+    })
+
+    return parsedError
```

Comment:
I'd just call `parsedError`, `err`. If you want to differentiate between the serialized arg, I'd call the arg, `errorString`. 🤷‍♂️

Reply:
looks like the error isn't always a string, so disregard that comment lol

Reply:
Agreed. What I am thinking about doing is moving the `deserializeCrossDomainErrorFromObject` contents over to `correctStackForCrossDomainError` and just handling `AssertionError` or `Error` (since we aren't sure what else is coming over yet) and just overwriting the stack. I think there is a bit more work to be done in the printed stack trace in the reporter, but I think that's due to error properties not getting assigned properly. I'm going to take a look into it and let you know how it goes!